packages/web/pages/_app.tsx

```diff
@@ -185,6 +191,39 @@ const MainLayoutWrapper: FunctionComponent<{ children: ReactNode }> = observer(
     return <MainLayout menus={menus}>{children}</MainLayout>;
   }
 );
+
+export const AppsBadge: FunctionComponent<{ appsLink: string }> = observer(
+  ({ appsLink }) => {
+    const { newApps, allApps } = useNewApps();
+    const router = useRouter();
+    const [viewedAppTitles, setViewedAppTitles] = useLocalStorageState<
+      string[]
+    >("viewed-apps", []);
+
+    useEffect(() => {
+      if (router.pathname === appsLink && allApps.length > 0) {
```

Comment:
can you put some comments here? I'm not exactly sure why you are doing it this way.

Reply:
I primarily did it to colocate the logic making it more straightforward. It also has the following benefits: 
1. It ensures that the viewed app titles are updated as soon as the user navigates to the apps page. Otherwise, we would need to rely on context, event emitters, or mobx to get the updated state within AppsBadge.
2. It reduces the complexity by not managing state in two files
3. It improves the performance of the apps page by avoiding re-renders whenever the viewed app titles change.